plugins/modules/system/pam_limits.py

```diff
@@ -289,16 +290,25 @@ def main():
     f.close()
     nf.flush()
 
-    # Copy tempfile to newfile
-    module.atomic_move(nf.name, f.name)
+    with open(limits_conf, 'r') as content:
+        content_current = content.read()
+
+    with open(nf.name, 'r') as content:
+        content_new = content.read()
+
+    if not module.check_mode:
+        # Copy tempfile to newfile
+        module.atomic_move(nf.name, f.name)
```

Comment:
It's probably better to use `limits_conf` here as well:
```suggestion
        module.atomic_move(nf.name, limits_conf)
```

Reply:
Done.